montepy/input_parser/input_file.py

```diff
@@ -32,6 +32,19 @@ def __init__(self, path, parent_file=None, overwrite=False):
         self._mode = None
         self._fh = None
 
+    @classmethod
+    def from_open_stream(cls, fh):
+        """
+        Create an MCNP Input File from an open, writable stream
+
+        :param fh: An open and writable object, such as a file handle.
+        :type fh: io.TextIOBase
+        """
+        name = getattr(fh, "name", fh.__class__.__name__)
```

Comment:
Would `type(fh).__name__` be more pythonic?

Reply:
I generally use `.__class__.__name__` for printing, to respect an overwrite of `.__class__` if that should occur, but have no strong feelings on the matter.

Reply:
Ok that's good enough for me.